### PR TITLE
:chore improve send flow + ui tests swaps / send 

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Description

* Show error message in first screen when user has insufficient funds for gas fees
* Add UI tests for all swap / send flows

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes [APKT-1407](https://linear.app/reown/issue/APKT-1407/swaps-not-working-as-expected-for-eoasa-accounts)
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/core/src/controllers/SendController.ts [hunk 1]
```diff
@@ -154,6 +160,54 @@ export const SendController = {
...
+    const gasPrice = gasPriceInUSD || '0'
+
+    if (NumberUtil.bigNumber(networkBalanceInUSD).isZero()) {
+      return true
+    }
+
+    return NumberUtil.bigNumber(NumberUtil.bigNumber(gasPrice)).isGreaterThan(networkBalanceInUSD)
+  },
+
+  hasInsufficientToken() {
```
Comment (magiziz) [inline packages/core/src/controllers/SendController.ts:193]: Is this insufficient gas or unsupported token ?

**Review by magiziz — APPROVED** [review]

File: packages/core/src/controllers/SendController.ts (hunk 1 above)
Comment (svenvoskamp) [reply packages/core/src/controllers/SendController.ts:193]: Gas funds, will change the name

File: packages/scaffold-ui/src/views/w3m-wallet-send-view/index.ts [hunk 2]
```diff
@@ -130,6 +133,10 @@ export class W3mWalletSendView extends LitElement {
       this.message = 'Add Address'
     }
 
+    if (SendController.hasInsufficientGasFunds()) {
+      this.message = 'Insufficient Gas Funds'
```
Comment (enesozturk) [inline packages/scaffold-ui/src/views/w3m-wallet-send-view/index.ts:137]: The message on the swaps prob is different, would it make sense to align them?

Comment (enesozturk) [reply packages/scaffold-ui/src/views/w3m-wallet-send-view/index.ts:137]: `Insufficient balance`

not sure which is best, we could keep different tho

**Review by enesozturk — APPROVED** [review]

**Pull request merged**